test/tests/functional/pbs_holidays.py

```diff
@@ -146,3 +170,117 @@ def test_only_year(self):
+
+        self.check_prime()
+
+        # Verify that scheduler didn't log a message for out of date file
+        msg = "holidays;The holiday file is out of date; please update it."
+        self.scheduler.log_match(msg, starttime=ctime, existence=False,
+                                 max_attempts=5)
+
+    def test_stale_year(self):
+        """
+        Test that the scheduler logs out-of-date log message and assumed
+        all prime-time for a holidays file with just a stale year line
+        """
+        ctime = int(time.time())
+        self.scheduler.holidays_delete_entry('a')
+
+        self.scheduler.holidays_set_year(self.cur_year - 1)
+
+        # Verify that the holidays file has only the year line
+        h_content = self.scheduler.holidays_parse_file()
+        self.assertEqual(len(h_content), 1)
+        self.assertEqual(h_content[0], "YEAR\t%d" % (self.cur_year - 1))
+
+        self.check_prime()
+
+        # Verify that scheduler logged a message for out of date file
+        msg = "holidays;The holiday file is out of date; please update it."
+        self.scheduler.log_match(msg, starttime=ctime)
+
+    def test_commented_holidays_file(self):
+        """
+        Test that the scheduler assumes all prime-time if the holidays
+        file is completely commented out
+        """
+        ctime = int(time.time())
+        self.scheduler.holidays_delete_entry('a')
+
+        content = """* YEAR 1970
+*  weekday 0600  1730
+*  saturday  none  all
+*  sunday  none  all"""
+
+        self.scheduler.holidays_write_file(content=content)
+
+        # Verify that the holidays file has 4 lines
+        h_content = self.du.cat(filename=self.scheduler.holidays_file,
+                                sudo=True)['out']
+        self.assertEqual(len(h_content), 4)
+
+        self.check_prime()
+
+        # Verify that scheduler didn't log a message for out of date file
+        msg = "holidays;The holiday file is out of date; please update it."
+        self.scheduler.log_match(msg, starttime=ctime, existence=False,
+                                 max_attempts=5)
+
+    def test_non_prime(self):
+        """
+        Test that non-prime time set via holidays file works correctly
+        """
+        self.scheduler.holidays_delete_entry('a')
+
+        self.scheduler.holidays_set_year(self.cur_year)
+
+        # Set all days of the week as prime time except today
+        today_idx = datetime.datetime.today().weekday()
+        today = self.days[today_idx]
+        for i in range(7):
+            if i != today_idx:
+                self.scheduler.holidays_set_day(self.days[i], "all", "none")
+
+        # Set today as all non-prime time
+        self.scheduler.holidays_set_day(today, "none", "all")
+        self.check_nonprime()
+
+    def test_missing_year(self):
+        """
+        Test that scheduler assumes all prime time if the year entry is
+        missing from holidays file
+        """
```

Comment:
You really have tested this above.  You are testing what the scheduler will do with a completely commented out holidays file.  Above, the year was comment out, and here the commented out year is missing.  In either case the entire file is commented out.

Reply:
I know, but I'd like to keep it just to be thorough.

Reply:
Then make the test different.  Set non-prime such that it should be in non-prime, but since the year is missing it will be 24hr prime.

Reply:
Actually I think the test already does that, it sets 24x7 non-prime time and then checks that it's 24x7 prime-time because of the missing year entry

Reply:
It has 3 lines that are commented out.  I'm saying don't have them commented out.  Have them set non-prime.  This way we can easily check if 24hr prime is working.

Reply:
D'oh! Those lines were never meant to commented out, sorry about that, yes, the test would be very much the same as the previous one with them commented out :) I've uncommented them now, please let me know if it looks good now.